spec/system/check_ahoy_tracking_spec.rb

```diff
@@ -2,192 +2,137 @@
   include_context 'with progress'
-
   include_context 'with user'
+  include AlphaAssessmentShared
 
-  context 'when a user has visited the first page of the module (after interruption page)' do
-    before do
-      start_module(alpha)
-    end
+  let(:alpha_event) do
+    Ahoy::Event.where(user_id: user.id).where_properties(training_module_id: 'alpha')
+  end
+  let(:bravo_event) do
+    Ahoy::Event.where(user_id: user.id).where_properties(training_module_id: 'bravo')
+  end
 
+  context 'when a user has visited the first page of the module (after interruption page)' do
     it 'ahoy tracks the starting of a module - track(module_start)' do
+      start_module(alpha)
       visit '/modules/alpha/content-pages/before-you-start'
       click_on 'Next'
       expect(page).to have_current_path '/modules/alpha/content-pages/intro', ignore_query: true
-      # check ahoy is tracking event
-      events = Ahoy::Event.where(user_id: user.id, name: 'module_start').where_properties(training_module_id: 'alpha')
-      expect(events.size).to eq 1
-    end
-  end
-
-  context 'when a user has finished the module (at the certificate page)' do
-    before do
-      view_pages_before_confidence_check(alpha)
-    end
-
-    it 'ahoy tracks the completion of a module - track(module_complete)' do
-      visit '/modules/alpha/content-pages/intro'
-      visit '/modules/alpha/content-pages/1-3-3'
-      click_on 'Next'
-      2.times do
-        check 'Correct answer 1'
-        check 'Correct answer 2'
-        click_on 'Next'
-      end
-      choose 'Correct answer 1'
-      click_on 'Next'
-      click_on 'Finish'
-      expect(page).to have_current_path '/modules/alpha/certificate', ignore_query: true
-      # check ahoy is tracking the event
-      events = Ahoy::Event.where(user_id: user.id, name: 'module_complete').where_properties(training_module_id: 'alpha')
-      expect(events.size).to eq 1
+      expect(alpha_event.where(name: 'module_start').size).to eq 1
     end
   end
 
   context 'when a user has visited the first page of the confidence check' do
-    before do
-      view_pages_before_confidence_check(alpha)
-    end
-
     it 'ahoy tracks the starting of the confidence check - track(confidence_check_start)' do
+      view_pages_before_confidence_check(alpha)
       visit '/modules/alpha/content-pages/1-3-3'
       click_on 'Next'
       expect(page).to have_current_path '/modules/alpha/confidence-check/1-3-3-1', ignore_query: true
-      # check ahoy is tracking the event
-      events = Ahoy::Event.where(user_id: user.id, name: 'confidence_check_start').where_properties(training_module_id: 'alpha')
-      expect(events.size).to eq 1
+      expect(alpha_event.where(name: 'confidence_check_start').size).to eq 1
     end
   end
 
   context 'when a user has answered a confidence check question' do
-    before do
+    it 'ahoy tracks when a confidence check question has been answered correctly - track(questionnaire_answer)' do
```

Comment:
NB: when updating to accommodate the refactored code we want to change this description.

`when a confidence questionnaire is answered`... it `is recorded in the event log`